aruco_ros/CMakeLists.txt

```diff
@@ -71,18 +78,25 @@ target_include_directories(double
 ament_target_dependencies(double ${THIS_PACKAGE_INCLUDE_DEPENDS})
 target_link_libraries(double ${OpenCV_LIBRARIES})
 
-add_executable(marker_publisher src/marker_publish.cpp
-                                src/aruco_ros_utils.cpp)
-target_include_directories(marker_publisher
+add_library(marker_publisher_component 
+  SHARED
+  src/marker_publish.cpp
+  src/aruco_ros_utils.cpp)
+
+target_include_directories(marker_publisher_component
   PUBLIC
   include)
 
-target_include_directories(marker_publisher
+target_include_directories(marker_publisher_component
   SYSTEM PUBLIC
   ${OpenCV_INCLUDE_DIRS}
 )
-ament_target_dependencies(marker_publisher ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(marker_publisher ${OpenCV_LIBRARIES})
+ament_target_dependencies(marker_publisher_component ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(marker_publisher_component ${OpenCV_LIBRARIES})
+
+rclcpp_components_register_node(marker_publisher_component
+  PLUGIN "aruco_ros::ArucoMarkerPublisherComponent"
+  EXECUTABLE marker_publisher)
```

Comment:
executable to maintain current behavior

Reply:
Yes, better to maintain current behaviour